examples/quantum1d/7.xy-finiteT/main.jl

```diff
@@ -0,0 +1,471 @@
+Then, we can write
+
+```math
+x^n = x^{2^m} = x^{2^{m-1}} \cdot x^{2^{m-1}} = (x^{2^{m-2}} \cdot x^{2^{m-2}}) \cdot (x^{2^{m-2}} \cdot x^{2^{m-2}}) = \dots
+```
+
+In other words, we can scan an exponential range of $\beta$ values by squaring the density matrix at each step.
+"""
+
+βs_exp = 2.0 .^ (-3:3)
+Z_mpo_mul_exp = zeros(length(βs_exp))
+
+## first iteration: start from high order Taylor expansion
+ρ₀ = make_time_mpo(H, im * first(βs_exp) / 2, TaylorCluster(; N=3))
+Z_mpo_mul_exp[1] = real(dot(ρ₀, ρ₀))^(1 / N)
+
+## subsequent iterations: square
+ρ = ρ₀
+ρ_mps = convert(FiniteMPS, ρ₀)
+for i in 2:length(βs_exp)
+    global ρ_mps, ρ
+    @info "Computing β = $(βs_exp[i])"
+    ρ_mps, = approximate(ρ_mps, (ρ, ρ_mps),
+                         DMRG2(; trscheme=truncdim(D_max), maxiter=10))
+    Z_mpo_mul_exp[i] = real(dot(ρ_mps, ρ_mps))^(1 / N)
+    ρ = convert(FiniteMPO, ρ_mps)
+end
+F_mpo_mul_exp = -(1 ./ βs_exp) .* log.(Z_mpo_mul_exp)
+
+p_mpo_mul_exp_diff = let
+    labels = reshape(map(expansion_orders[2:end]) do N
+                         return "Taylor N=$N"
+                     end, 1, :)
+    p1 = plot(βs, abs.(real.(Z_taylor2) .- partition_function.(βs, J, N));
+              label=labels, title="Partition function error",
+              xlabel="β", ylabel="ΔZ(β)")
+    plot!(p1, βs, abs.(real.(Z_mpo_mul) .- partition_function.(βs, J, N));
+          label="MPO multiplication")
+    plot!(p1, βs_exp, abs.(real.(Z_mpo_mul_exp) .- partition_function.(βs_exp, J, N));
+          label="MPO multiplication exp")
+
+    p2 = plot(βs, abs.(real.(F_taylor2) .- free_energy.(βs, J, N)); label=labels,
+              xlabel="β", ylabel="ΔF(β)", title="Free energy error")
+    plot!(p2, βs, abs.(real.(F_mpo_mul) .- free_energy.(βs, J, N));
+          label="MPO multiplication")
+    plot!(p2, βs_exp, abs.(real.(F_mpo_mul_exp) .- free_energy.(βs_exp, J, N));
+          label="MPO multiplication exp")
+    plot(p1, p2)
+end
+
+md"""
+Clearly, the exponential approach allows us to reach larger $\beta$ values much quicker, but there is again a trade-off.
+Since the size of the steps are increasing, we need to be more careful with the accuracy of our approximations.
+
+!!! warning
+    Again, using MPS techniques to approximate the multiplication of density matrices might lead to unphysical truncated density matrices.
+    Increasing the stepsize could make this happen sooner, so we need to be careful with the maximal bond dimension.
+"""
+
+md"""
+### Time evolution approach
+
+Finally, we can also note that the partition function is characterized by the following differential equation:
+
+```math
+\frac{dZ}{d\beta} = -H \cdot Z
+\implies Z(\beta) = e^{-\beta H} \cdot Z(0)
+```
+
+In other words, we can compute the partition function at $\beta$ by evolving the partition function at $0$ for a time $d\tau = -im \beta$.
```

Comment:
```suggestion
In other words, we can compute the partition function at $\beta$ by evolving the partition function at $0$ for a time $d\tau = -i \beta$.
```
Latex does not speak Julia :)